pkg/karmadactl/cmdinit/utils/util.go

```diff
@@ -18,13 +18,20 @@ type Downloader struct {
 	Current int64
 }
 
+// only set true in test
+var isTesting = false
+
 // Read Implementation of Downloader
 func (d *Downloader) Read(p []byte) (n int, err error) {
 	n, err = d.Reader.Read(p)
 	d.Current += int64(n)
-	fmt.Printf("\rDownloading...[ %.2f%% ]", float64(d.Current*10000/d.Total)/100)
-	if d.Current == d.Total {
-		fmt.Printf("\nDownload complete.")
+	if !isTesting {
```

Comment:
Why don't we just make the change as shown in https://github.com/karmada-io/karmada/issues/2671#issuecomment-1287828359?

I suppose the real code should not involve testing logic.

Reply:
I don't known what the affect of this change. So I keep the origin format.

Reply:
It should be OK. the only effect I can imagine is the output messages might add an additional `blank` line(`\n`).

Reply:
```go
func main() {
	fmt.Printf("\rDownloading...[ %.2f%% ]", 90.0)
	fmt.Printf("\rDownloading...[ %.2f%% ]", 95.0)
	fmt.Printf("\rDownloading...[ %.2f%% ]", 100.0)
	fmt.Printf("\nDownload complete.")
}
```
Output:
```
Downloading...[ 100.00% ]
Download complete.
```

```go
func main() {
	fmt.Printf("\rDownloading...[ %.2f%% ]\n", 90.0)
	fmt.Printf("\rDownloading...[ %.2f%% ]\n", 95.0)
	fmt.Printf("\rDownloading...[ %.2f%% ]\n", 100.0)
	fmt.Printf("Download complete.\n")
}
```

Output:
```
Downloading...[ 90.00% ]
Downloading...[ 95.00% ]
Downloading...[ 100.00% ]
Download complete.
```

Reply:
These are different. In origin code the `Downloading ...` log will cover the previous.

Reply:
@RainbowMango , I have remove the `isTesting` flag. Instead add a `io.Writer` output, and print the log to it. In tests, output is assigned `ioutil.Discard`, and will not print the log.

---
I test this code before change, and output:
```
I1026 14:28:17.690860   50020 deploy.go:269] download crds file:https://github.com/karmada-io/karmada/releases/download/v1.3.0/crds.tar.gz
Downloading...[ 100.00% ]
Downloading...[ 100.00% ]
Download complete.e
```

after change:
```
I1026 14:31:01.082274   50274 deploy.go:269] download crds file:https://github.com/karmada-io/karmada/releases/download/v1.3.0/crds.tar.gz
Downloading...[ 100.00% ]
Downloading...[ 100.00% ]
Download complete.
```

These outputs is same.